diff --git a/etc/warewulf.conf b/etc/warewulf.conf
index ab06c8b0..442faf18 100644
--- a/etc/warewulf.conf
+++ b/etc/warewulf.conf
@@ -1,5 +1,5 @@
-ipaddr: 10.0.0.1
-netmask: 255.255.252.0
+ipaddr: 10.0.0.2
+netmask: 255.255.255.0
 warewulf:
   secure: false
   update interval: 60
@@ -7,8 +7,8 @@ warewulf:
   host overlay: true
 dhcp:
   enabled: true
-  range start: 10.0.1.1
-  range end: 10.0.1.255
+  range start: 10.0.0.200
+  range end: 10.0.0.255
   systemd name: dhcpd
   template: default
 tftp:
@@ -24,8 +24,8 @@ nfs:
   export paths:
   - path: /home
     export options: rw,sync
-  - path: /opt
-    export options: ro,sync,no_root_squash
+#  - path: /opt
+#    export options: ro,sync,no_root_squash
   systemd name: nfs-server
 image mounts:
   - source: /etc/resolv.conf
diff --git a/internal/pkg/overlay/overlay.go b/internal/pkg/overlay/overlay.go
index 3b76ae4f..9142db15 100644
--- a/internal/pkg/overlay/overlay.go
+++ b/internal/pkg/overlay/overlay.go
@@ -368,6 +368,12 @@ func BuildOverlayIndir(nodeData node.Node, allNodes []node.Node, overlayNames []
 					softlinkFromTemplate := regLink.FindAllStringSubmatch(line, -1)
 					if len(softlinkFromTemplate) != 0 {
 						wwlog.Debug("Creating soft link %s -> %s", outputPath, softlinkFromTemplate[0][1])
+						if _, err := os.Lstat(filePath); err == nil {
+							wwlog.Debug("Link %s already exists, will try to remove", filePath)
+							if err := os.Remove(filePath); err != nil {
+								return fmt.Errorf("failed to remove existing link at %s: %w", filePath, err)
+							}
+						}
 						return os.Symlink(softlinkFromTemplate[0][1], outputPath)
 					} else if len(filenameFromTemplate) != 0 {
 						wwlog.Debug("Writing file %s", filenameFromTemplate[0][1])
@@ -447,6 +453,13 @@ Writes buffer to the destination file. If wwbackup is set a wwbackup will be cre
 */
 func CarefulWriteBuffer(destFile string, buffer bytes.Buffer, backupFile bool, perm fs.FileMode) error {
 	wwlog.Debug("Trying to careful write file (%d bytes): %s", buffer.Len(), destFile)
+	parentDir := path.Dir(destFile)
+    if !util.IsDir(parentDir) {
+        wwlog.Debug("Creating parent directory: %s", parentDir)
+        if err := os.MkdirAll(parentDir, 0o755); err != nil {
+            return fmt.Errorf("failed to create parent directories for %s: %w", destFile, err)
+        }
+    }
 	if backupFile {
 		if !util.IsFile(destFile+".wwbackup") && util.IsFile(destFile) {
 			err := util.CopyFile(destFile, destFile+".wwbackup")
diff --git a/overlays/host/rootfs/etc/dhcp/dhcpd.conf.ww b/overlays/host/rootfs/etc/dhcp/dhcpd.conf.ww
index fecb1fed..423af76c 100644
--- a/overlays/host/rootfs/etc/dhcp/dhcpd.conf.ww
+++ b/overlays/host/rootfs/etc/dhcp/dhcpd.conf.ww
@@ -54,6 +54,7 @@ subnet {{.Network}} netmask {{.Netmask}} {
 {{- end }}
 {{- if .Ipaddr }}
     next-server {{.Ipaddr}};
+    option tftp-server-name "{{.Ipaddr}}";
 {{- end }}
 }
 {{- end }}
